### PR TITLE
added edit mode to compare over time

### PR DESCRIPTION
**Pull request by Carla-Moz**
This PR adds the edit mode to compare over time, along with the read only mode for the selected base repo and time range text. 

Closes [Compare over time: edit feature for results page ](https://mozilla-hub.atlassian.net/browse/PCF-405)

Screenshot 2024-06-11 at 13 53 54

Screenshot 2024-06-11 at 13 54 02

### CODE REVIEW

File: src/components/Search/SearchOverTime.tsx [hunk 1]
```diff
@@ -82,12 +81,23 @@ export default function SearchOverTime({
...
+    padding: `${Spacing.xSmall}px ${Spacing.xSmall}px`,
+  });
+
   return (
     <Grid className={styles.component}>
       {/**** Base - Time-Range Labels ****/}
       <Grid
         container
         spacing={2}
+        mb={0.5}
```
Comment (julienw) [inline src/components/Search/SearchOverTime.tsx:100]: This added margin looks wrong. Indeed on my computer it seems to be too much, and especially it seems be counted in addition to the other margins, and therefore the visual margin below "Base repository" is wider than the one Below "Revisions".

**Review by julienw — APPROVED** [review]
Thanks, this looks good to me in terms of behavior. Just I think using socalled "conditional rendering" instead of the "display" value would be better here. In general in React-based programming I find this easier to comprehend when looking at the code. (Note: in more vanilla-based JS I'd do it more with classes like you did).

I see a few style-related issues, that I think can be handled in a next PR:
* margins (or paddings) could be improved. Especially the form doesn't look balanced: the margin between the first row (base repo and time range) and the second row (Revisions) is too small -- especially compared to the margin between "Revisions" and "Framework", and to the "Compare with base" form.
* the margins between the title of a row and the input are not consistent either, when looking at edit mode vs non-edit mode, and when comparing the rows with each other.
* When clicking "Edit entry", because that button disappears completely, everything moves up. I think it is a good case of using `visibility: hidden` on that "Edit entry" button, so that it take up the same amount of space than when it's present. It's not moving like that in "Compare with base" edit mode, either by chance or because you designed the margins finely? :-)

File: package-lock.json
[the remark this answers is not in the document]
Comment (Carla-Moz) [reply package-lock.json:1]: Yeah, that was unintentional. Will do.

File: src/components/Search/SearchOverTime.tsx (hunk 1 above)
Comment (Carla-Moz) [reply src/components/Search/SearchOverTime.tsx:100]: I'll do a styles cleanup PR next.

File: src/components/Search/SearchOverTime.tsx [hunk 2]
```diff
@@ -82,12 +81,23 @@ export default function SearchOverTime({
     },
   });
 
+  const timeRangeText = timeRanges.find(
+    (entry) => entry.value === timeRangeValue,
+  )?.text;
+
+  const readOnlyStyles = style({
+    borderRadius: Spacing.xSmall,
+    backgroundColor: Colors.Background200,
```
[the remark this answers is not in the document]
Comment (Carla-Moz) [reply src/components/Search/SearchOverTime.tsx:90]: Yes, it does. I'll clean it up in a new PR.

**Comment by Carla-Moz** [comment]
@julienw, thanks for the review! I'll address the design bugs in a new PR.

**Pull request merged**